src/framework/mpas_dmpar.F

```diff
@@ -6167,6 +6183,201 @@ subroutine mpas_dmpar_exch_halo_field5d_real(field, haloLayersIn)!{{{
+                      nAdded = nAdded + 1
+                    end do
+                  end do
+                end if
+
+                exchListPtr => exchListPtr % next
+              end do
+
+              fieldCursor => fieldCursor % next
+            end do
+            bufferOffset = bufferOffset + nAdded
+          end do
+
+          call MPI_Isend(commListPtr % rbuffer, commListPtr % nList, MPI_REALKIND, commListPtr % procID, dminfo % my_proc_id, dminfo % comm, commListPtr % reqID, mpi_ierr)
+          commListPtr => commListPtr % next
+        end do
+#endif
+
+        ! Handle local copy. If MPI is off, then only local copies are performed.
+        fieldCursor => field
+        do while(associated(fieldCursor))
+          do iHalo = 1, nHaloLayers
+            exchListPtr => fieldCursor % copyList % halos(haloLayers(iHalo)) % exchList
+
+            do while(associated(exchListPtr))
+              fieldCursor2 => field
+              do while(associated(fieldCursor2))
+                if(exchListPtr % endPointID == fieldCursor2 % block % localBlockID) then
+                  call mpas_log_write(' -- exch_halo_adj_field2d  local exchListPtr % nList:$i ',&
+                     intArgs=(/exchListPtr % nList/))
+                  do i = 1, exchListPtr % nList
+                    !fieldCursor2 % array(:, exchListPtr % destList(i)) = fieldCursor % array(:, exchListPtr % srcList(i))
+                    fieldCursor % array(:, exchListPtr % srcList(i)) = fieldCursor % array(:, exchListPtr % srcList(i)) + fieldCursor2 % array(:, exchListPtr % destList(i))
+                    fieldCursor2 % array(:, exchListPtr % destList(i)) = 0.0_RKIND
+                  end do
+                end if
+
+                fieldCursor2 => fieldCursor2 % next
+              end do
+
+              exchListPtr => exchListPtr % next
+            end do
+          end do
+
+          fieldCursor => fieldCursor % next
+        end do
+
+#ifdef _MPI
+
+        ! Wait for mpi_irecv to finish, and unpack data from buffer
+        commListPtr => sendList
+        do while(associated(commListPtr))
+          call MPI_Wait(commListPtr % reqID, MPI_STATUS_IGNORE, mpi_ierr)
+          bufferOffset = 0
+          do iHalo = 1, nHaloLayers
+            nAdded = 0
+            fieldCursor => field
+            do while(associated(fieldCursor))
+              exchListPtr => fieldCursor % sendList % halos(haloLayers(iHalo)) % exchList
+              do while(associated(exchListPtr))
+                if(exchListPtr % endPointID == commListPtr % procID) then
+                  call mpas_log_write(' -- exch_halo_adj_field2d  unpack exchListPtr % nList:$i fieldCursor % dimSizes:$i',&
+                     intArgs=(/exchListPtr % nList, fieldCursor % dimSizes(1)/))
+                  do i = 1, exchListPtr % nList
+                    if (present(interiorCells)) then
+                       if (interiorCells(exchListPtr % srcList(i)) .eq. 1) then
```

Comment:
I saw this print statement when running the tests, which indicates interior cells were getting updated via the halo adjoint exchange.